cli/local/delete.go

```diff
@@ -19,29 +19,58 @@ package local
 import (
 	"fmt"
 
+	"github.com/cortexlabs/cortex/pkg/consts"
 	"github.com/cortexlabs/cortex/pkg/lib/docker"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/prompt"
 	"github.com/cortexlabs/cortex/pkg/operator/schema"
 	"github.com/cortexlabs/cortex/pkg/types/spec"
 )
 
-func Delete(apiName string, keepCache bool) (schema.DeleteResponse, error) {
+func Delete(apiName string, keepCache, deleteForce bool) (schema.DeleteResponse, error) {
 	_, err := docker.GetDockerClient()
 	if err != nil {
 		return schema.DeleteResponse{}, err
 	}
 
 	var apiSpec *spec.API = nil
-	if !keepCache {
-		if apiSpec, err = FindAPISpec(apiName); err != nil {
+	if apiSpec, err = FindAPISpec(apiName); err != nil {
+		if errors.GetKind(err) == ErrCortexVersionMismatch {
+			var incompatibleVersion string
+			if incompatibleVersion, err = GetVersionFromAPISpec(apiName); err != nil {
+				return schema.DeleteResponse{}, err
+			}
+			if !deleteForce {
+				prompt.YesOrExit(
+					fmt.Sprintf(
+						"api %s was deployed using CLI version %s but the current CLI version is %s; "+
+							"deleting %s with current CLI version %s might break the CLI; any cached models won't be deleted\n\n"+
```

Comment:
I'm thinking instead of `download an older version of the CLI using github.com/cortexlabs/cortex/blob/VERSION/get-cli.sh, delete the API using the older version of the CLI and then redeploy the API using the latest version of the CLI` -> `it is recommended to download version %s of the CLI from https://docs.cortex.dev/v/<minor-mismatched-version>/install, delete the API using version <minor-mistmatched-version> of the CLI and then re-deploy the API using the latest version of the CLI`.

This can happen regardless of whether the version the API has been deployed with is newer or older. And I think specifying the link to the documentation for installing the CLI is better. Same as for the other comment down below. Thoughts?

Reply:
Yes, you are correct, Ship it!